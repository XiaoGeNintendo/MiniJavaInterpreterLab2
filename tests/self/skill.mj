class Skill{
    string name;
    Skill(string name){
        this.name=name;
    }
    void effect(){

    }
}

class FireBall extends Skill{
    FireBall(){
        super("Fireball");
    }

    void effect(){
        println("Fireball!");
    }
}

class IceSpike extends Skill{
    IceSpike(){
        super("IceSpike");
    }

    void effect(){
        println("IceSpike!");
    }
}

class DragonDance extends Skill{
    DragonDance(){
        super("DragonDance");
    }

    void effect(){
        println("DragonDance!");
    }
}

void select(Skill[] sk, int id){
    println("You used "+sk[id-1].name+"!");
    sk[id-1].effect();
}

int main(){
    Skill[] sk=new Skill[]{new DragonDance(),new DragonDance(), new FireBall(),new IceSpike()};

    select(sk,1);
    select(sk,1);
    select(sk,3);
    select(sk,4);
    select(sk,2);
    select(sk,4);
    select(sk,3);
    return 0;
}
