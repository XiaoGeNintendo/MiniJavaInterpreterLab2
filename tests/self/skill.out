You used DragonDance!
DragonDance!
You used DragonDance!
DragonDance!
You used Fireball!
Fireball!
You used IceSpike!
IceSpike!
You used DragonDance!
DragonDance!
You used IceSpike!
IceSpike!
You used Fireball!
Fireball!
Process exits with the code 0.
